File: src/chan/content/model/Vote.java
package chan.content.model;

import chan.annotation.Public;

@Public
public final class Vote {

    private final int likes;
    private final int dislikes;
    private final boolean showVotes;

    @Public
    public Vote() {
        this.likes = 0;
        this.dislikes = 0;
        this.showVotes = false;
    }

    @Public
    public Vote(int likes, int dislikes) {
        this.likes = likes;
        this.dislikes = dislikes;
        this.showVotes = true;
    }

    @Public
    public int getLikes() {
        return likes;
    }

    @Public
    public int getDislikes() {
        return dislikes;
    }

    @Public
    public boolean isShowVotes() {
        return showVotes;
    }

}

File: src/chan/content/model/Vote.kt
package chan.content.model

import chan.annotation.Public

@Public
class Vote {
    private val likes: Int
    private val dislikes: Int
    private val showVotes: Boolean

    @Public
    constructor() {
        likes = 0
        dislikes = 0
        showVotes = false
    }

    @Public
    constructor(likes: Int, dislikes: Int) {
        this.likes = likes
        this.dislikes = dislikes
        showVotes = true
    }

    @Public
    fun getLikes(): Int = likes

    @Public
    fun getDislikes(): Int = dislikes

    @Public
    fun isShowVotes(): Boolean = showVotes
}
